File: frontend/src/types/blog.ts
export interface User {
  uid: string;
  email: string;
  displayName: string;
  joinedAt: Date;
}

export interface Blog {
  _id: string;
  title: string;
  content: string;
  tags: string[];
  authorId: string;
  author?: {
    displayName: string;
    email: string;
  };
  createdAt: Date;
  updatedAt: Date;
  isPublic: boolean;
}

export interface CreateBlogRequest {
  title: string;
  content: string;
  tags: string[];
  isPublic: boolean;
}

export interface UpdateBlogRequest extends CreateBlogRequest {
  _id: string;
}

File: backend/models/blog.py
# backend/models/blog.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

class BlogCreate(BaseModel):
    title: str
    content: str
    isPublic: bool = True
    tags: Optional[List[str]] = []

class BlogUpdate(BaseModel):
    title: Optional[str] = None
    content: Optional[str] = None
    isPublic: Optional[bool] = None
    tags: Optional[List[str]] = None

class BlogResponse(BaseModel):
    id: str
    title: str
    content: str
    authorId: str
    createdAt: datetime
    updatedAt: Optional[datetime] = None
    isPublic: bool
    tags: Optional[List[str]] = []
